### ISSUE TITLE
FileManager: FileManager crashes when deleting a file in a dotdirectory with a delete action.

### ISSUE THREAD

**Issue by emanuele6** [SerenityOS/serenity#3418]
To reproduce, try deleting `~/.config/Browser.ini` with the delete action in the toolbar.
```
USERSPACE(24) ASSERTION FAILED: index.is_valid()
../Libraries/LibGUI/ModelSelection.cpp:50
[FileManager(24:24)]: Terminating FileManager(24) due to signal 6
[FinalizerTask(3:3)]: 0xdeadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) +181
0xdeadc0de  Kernel::Scheduler::context_switch(Kernel::Thread*) +183
0xdeadc0de  Kernel::Scheduler::pick_next() +1681
0xdeadc0de  Kernel::Processor::check_invoke_scheduler() +125
0xdeadc0de  Kernel::Thread::die_if_needed() +157
0xdeadc0de  syscall_handler +1414
0xdeadc0de  syscall_asm_entry +49
0x080db1c3  raise +22
0x080480bd   +13
0x080d6db2  tolower +0
0x0807edc5  GUI::ModelSelection::set(GUI::ModelIndex const&) +41
0x0804e071  run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::{lambda()#2}::operator()() const +249
0x0804f109  run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::{lambda(run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::ConfirmBeforeDelete, GUI::Action const&)#13}::operator()(run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::ConfirmBeforeDelete, GUI::Action const&) const [clone .constprop.0] +595
0x0805be2e  GUI::Action::activate(Core::Object*) +172
0x0809eadc  WindowClientEndpoint::handle(IPC::Message const&) +256
0x0806c4b4  IPC::ServerConnection<WindowClientEndpoint, WindowServerEndpoint>::handle_messages() +106
0x080d4e22  Core::Object::dispatch_event(Core::Event&, Core::Object*) +74
0x080cdd9c  Core::EventLoop::pump(Core::EventLoop::WaitMode) +308
0x080cdf1b  Core::EventLoop::exec() +43
0x08051c89  run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String) +9550
0x08048bba  main +1967
0x08049a8e  _start +94
```

**Comment by emanuele6** [comment]
I don't know what fixed this, but it's fixed now.

**Comment by emanuele6** [comment]
Fixed by ef50e5aaeef0406e9e4fdeb8fceb92eb2e35c679.
